Header header
uint32 size
AgentMsg[] agents